File: packages/lsimons-$project-py/src/lsimons_$project/greet.py
"""Core library logic. Kept separate from any CLI so it is reusable."""


def greet(name: str) -> str:
    """Return ``hello, {name}``. Raises ``ValueError`` if ``name`` is empty."""
    if not name:
        raise ValueError("name must not be empty")
    return f"hello, {name}"

File: packages/lsimons-$project-go/greet.go
// Package greet is the Go implementation of the shared greet library.
package greet

import "errors"

// ErrEmptyName is returned by Greet when name is empty.
var ErrEmptyName = errors.New("name must not be empty")

// Greet returns "hello, <name>".
func Greet(name string) (string, error) {
	if name == "" {
		return "", ErrEmptyName
	}
	return "hello, " + name, nil
}
